frontend/src/components/ImplementationReportView/EmbedBadges.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from "react";
+import { Implementation } from "../../data/parseReportData";
+import CopyToClipboard from "../CopyToClipboard";
+import Dialect from "../../data/Dialect";
+import { complianceBadgeFor, versionsBadgeFor } from "../../data/Badge";
+import { badgeFormatOptions, BadgeFormatOption } from "./BadgeFormats";
+
+const EmbedBadges: React.FC<{
+  implementation: Implementation;
+}> = ({ implementation }) => {
+  const [activeTab, setActiveTab] = useState("URL");
+  const [activeBadge, setActiveBadge] = useState("JSON Schema Versions");
+  const [badgeURI, setBadgeURI] = useState(
+    versionsBadgeFor(implementation).href(),
```

Comment:
Convert to URI as late as possible (i.e. just before you actually render something), there's no reason to do it here I don't think.